src/Stratis.SmartContracts.Tests.Common/MockChain/MockChainNode.cs

```diff
@@ -177,8 +177,14 @@ public Result<WalletSendTransactionModel> SendTransaction(Script scriptPubKey, M
                 Sender = sender ?? this.MinerAddress.Address,
                 WalletName = this.WalletName
             };
-            JsonResult response = (JsonResult)this.smartContractsController.BuildAndSendCreateSmartContractTransaction(request);
-            return (BuildCreateContractTransactionResponse)response.Value;
+
+            IActionResult result = this.smartContractsController.BuildAndSendCreateSmartContractTransaction(request);
+            if (result is JsonResult response)
```

Comment:
Hmm what's this change for?

Reply:
Test assumes that result will always be of type JsonResult, however, we are sending a BadRequest response which is of a different type but uses the same IAsyncResult interface.

Reply:
Ah okay, I think this is breaking the Sct tool too

Reply:
We need to look into this more, make sure that Sct and the wallet can handle errors / bad request. Atm I think it's expecting errors to be output in "Message" inside the json or something similar

Reply:
Wallet is definitely fine as it handles bad requests from server. I will check SC tool

Reply:
Sct also handles ErrorResponses: https://github.com/stratisproject/Stratis.SmartContracts.Tools.Sct/blob/master/Stratis.SmartContracts.Tools.Sct/Deployment/HttpContractDeployer.cs